File: Irrgarten_Java/src/irrgarten/Shield.java
package irrgarten;

/**
 * Clase que representa los escudos que utiliza el jugador cuando se defiende
 * de un ataque de un monstruo.
 * 
 * @author Arturo Olivares Martos
 * @author Joaquín Avilés de la Fuente
 */
public class Shield extends CombatElement{
    
    /**
     * Constructor de la clase Shield.
     * @param protection La protección del escudo.
     * @param uses El número de usos disponibles del escudo.
     */
    public Shield(float protection, int uses){
        super(protection, uses);
    }
      
    /**
     * Método para realizar una protección con el escudo.
     * @return La intensidad de la protección del escudo.
     */
    public float protect(){
        return this.produceEffect();
    }
    
    /**
     * Método que devuelve una representación en forma de cadena de caracteres
     * del estado interno del objeto.
     * @return Representación en forma de cadena de caracteres del estado interno del objeto.
     */
    @Override
    public String toString() {
        String toReturn="S";
        toReturn+=super.toString();
        return toReturn;
    }
    
}

File: Irrgarten_Ruby/shield.rb
# encoding: UTF-8

require_relative 'combat_element'


module Irrgarten

    # Esta clase representa los escudos que utiliza el jugador en las defensas
    # durante los combates.
    #
    # @author Joaquin Avilés de la Fuente
    # @author Arturo Olivares Martos
    class Shield < CombatElement

        # Método que devuelve la protección del escudo y decrementa el número de usos en uno.
        #
        # @see CombatElement#produce_effect
        #
        # @return [float] Si tiene algún uso disponible devuelve la protección del escudo. En caso contrario (uses=0) devuelve 0
        def protect
            produce_effect
        end

        # Método que muestra en una cadena el estado del escudo, en cuanto a
        # uso y protección
        #
        # @return [String] devuelve una cadena que muestra usos y protección del escudo
        def to_s
            return "S" + super
        end

    end # class
end # module
